engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/date/NewNode.java

```diff
@@ -0,0 +1,32 @@
+package org.enso.interpreter.node.expression.builtin.date;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.nodes.Node;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.error.PanicException;
+
+@BuiltinMethod(
+    type = "Date_Internal",
+    name = "date_new",
+    description = "Constructs a new Date from a year, month, and day")
+public abstract class NewNode extends Node {
+  static NewNode build() {
+    return NewNodeGen.create();
+  }
+
+  abstract Object execute(Object self, long year, long month, long day);
+
+  @Specialization
+  Object doNew(Object self, long year, long month, long day) {
```

Comment:
You can save a ton of boilerplate by using Hubert's DSL – see `EnsoFile`

Reply:
More Hubert's DSL as of [a356457](https://github.com/enso-org/enso/pull/3559/commits/a356457e24c1cd44448e5a71a2887377f4c2c7c2)